### PR TITLE
docs(mpc): add mpc document

### PR DESCRIPTION
**Pull request by TakaHoribe**
## Description

Document for users who are not familiar with the MPC formulation.
This is the first version, not completed.

## Related links

None

## Tests performed

None

## Notes for reviewers

None

## Interface changes

None

## Effects on system behavior

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by takayuki5168 — APPROVED** [review]

File: control/mpc_lateral_controller/docs/mpc-algorithm-document.md
Comment (maxime-clem) [inline control/mpc_lateral_controller/docs/mpc-algorithm-document.md:1]: The new page shows up in a new `Docs` section. Is it okay ?
image

Comment (maxime-clem) [inline control/mpc_lateral_controller/docs/mpc-algorithm-document.md:1]: The equations don´t seem to work :thinking: 
https://autowarefoundation.github.io/autoware.universe/pr-4011/control/mpc_lateral_controller/docs/mpc-algorithm-document/

File: control/mpc_lateral_controller/docs/mpc-algorithm-document.md [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+
+Since the weight matrix can be added linearly, the final weight can be set as $Q = Q_{1} + Q_{2}$.
+
+Furthermore, MPC optimizes over a period of time, the time-varying weight can be considered in the optimization.
+
+### Constraints
+
+#### Input constraint
+
+The main advantage of the MPC controllers is the capability to deal with any state or input constraints. The constraints can be expressed as box constraints, such as that "tire angle must be within ±30 degrees" can be put in the form;
```
Comment (maxime-clem) [inline control/mpc_lateral_controller/docs/mpc-algorithm-document.md:330]:
```suggestion
The main advantage of MPC controllers is the capability to deal with any state or input constraints. The constraints can be expressed as box constraints, such as "the tire angle must be within ±30 degrees", and can be put in the following form;
```

**Review by maxime-clem — APPROVED** [review]
It looks great :+1:

File: control/mpc_lateral_controller/docs/mpc-algorithm-document.md
Comment (TakaHoribe) [reply control/mpc_lateral_controller/docs/mpc-algorithm-document.md:1]: Oops. I'll fix it. Thank you!

Comment (TakaHoribe) [reply control/mpc_lateral_controller/docs/mpc-algorithm-document.md:1]: Now it is like this

Comment (TakaHoribe) [reply control/mpc_lateral_controller/docs/mpc-algorithm-document.md:1]: Fixed.

**Pull request merged**